src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -17834,4 +17832,366 @@ static class E
+
+            // (30,21): error CS0663: 'Extensions' cannot define an overloaded method that differs only on parameter modifiers 'ref' and 'in'
+            //         public void M3() {}
+            Diagnostic(ErrorCode.ERR_OverloadRefKind, "M3").WithArguments("Extensions", "method", "ref", "in").WithLocation(30, 21),
+            // (40,21): error CS0663: 'Extensions' cannot define an overloaded method that differs only on parameter modifiers 'ref' and 'ref readonly'
+            //         public void M4() {}
+            Diagnostic(ErrorCode.ERR_OverloadRefKind, "M4").WithArguments("Extensions", "method", "ref", "ref readonly").WithLocation(40, 21),
+            // (50,21): error CS0663: 'Extensions' cannot define an overloaded method that differs only on parameter modifiers 'in' and 'ref readonly'
+            //         public void M5() {}
+            Diagnostic(ErrorCode.ERR_OverloadRefKind, "M5").WithArguments("Extensions", "method", "in", "ref readonly").WithLocation(50, 21),
+            // (59,24): error CS0663: 'Extensions' cannot define an overloaded method that differs only on parameter modifiers 'ref' and 'in'
+            //     static public void M7(this ref int receiver) {}
+            Diagnostic(ErrorCode.ERR_OverloadRefKind, "M7").WithArguments("Extensions", "method", "ref", "in").WithLocation(59, 24),
+            // (63,24): error CS0663: 'Extensions' cannot define an overloaded method that differs only on parameter modifiers 'ref' and 'ref readonly'
+            //     static public void M8(this ref int receiver) {}
+            Diagnostic(ErrorCode.ERR_OverloadRefKind, "M8").WithArguments("Extensions", "method", "ref", "ref readonly").WithLocation(63, 24),
+            // (67,24): error CS0663: 'Extensions' cannot define an overloaded method that differs only on parameter modifiers 'in' and 'ref readonly'
+            //     static public void M9(this in int receiver) {}
+            Diagnostic(ErrorCode.ERR_OverloadRefKind, "M9").WithArguments("Extensions", "method", "in", "ref readonly").WithLocation(67, 24),
+            // (72,21): error CS0111: Type 'Extensions.extension(object)' already defines a member called 'M10' with the same parameter types
+            //         public void M10() {}
+            Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "M10").WithArguments("M10", "Extensions.extension(object)").WithLocation(72, 21)
+            );
+
+        comp = CreateCompilation("""
+static class Extensions
+{
+    extension(int receiver)
+    {
+        public void M2() {}
+    }
+
+    extension(ref int receiver)
+    {
+        public void M2() {}
+    }
+
+    extension(int receiver)
+    {
+        public void M11() {}
+    }
+
+    extension(string receiver)
+    {
+        public void M11() {}
+    }
+}
+""");
+
+        CompileAndVerify(comp).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void SignatureConflict_02()
```

Comment:
For static scenarios, consider including some modifiers differences on the extension parameter too. That should not affect the result

Reply:
Consider adding a test for the factory scenario:
```
extension(C)
{
  static C Create() ...
}
extension(D)
{
  static D Create() ...
}
```